subsys/net/lib/openthread/platform/radio.c

```diff
@@ -361,12 +362,49 @@ otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
 	return &sTransmitFrame;
 }
 
+static void get_rssi_energy_detected(struct device *dev, s16_t max_ed)
+{
+	ARG_UNUSED(dev);
+	ot_signal_set(OPENTHREAD_SIGNAL_BLOCKING, max_ed);
+}
+
 int8_t otPlatRadioGetRssi(otInstance *aInstance)
 {
-	ARG_UNUSED(aInstance);
+	s8_t ret_rssi = INT8_MAX;
+	int signal_value;
+
+	otRadioCaps radioCaps =  otPlatRadioGetCaps(aInstance);
+
+	if (!(radioCaps & OT_RADIO_CAPS_ENERGY_SCAN)) {
+		/*
+		 *TODO: No API in Zephyr to get the RSSI
+		 * when OT_RADIO_CAPS_ENERGY_SCAN is not available
+		 */
+		ret_rssi = 0;
+	} else {
+		/*
+		 * Blocking implementation of get RSSI
+		 * using no-blocking ed_scan
+		 */
+		int error = 0;
+		const u16_t energy_detection_time = 1;
+
+		error = radio_api->ed_scan(radio_dev, energy_detection_time,
+					    get_rssi_energy_detected);
```

Comment:
I was more thinking about dedicated mutex instead of reusing existing events. I'm not sure if reusing existing event for different tasks won't get us into trouble in the future.

Reply:
It is safe because OPENTHREAD_SIGNAL_BLOCKING will be reserved only to block calls from OT,
and for OT, it is possible to have only one blocking call at the moment